src/announcements/react.js

```diff
@@ -0,0 +1,249 @@
+// @flow
+import uniq from "lodash/uniq";
+import intersection from "lodash/intersection";
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useReducer,
+} from "react";
+import type { Announcement } from "./types";
+import { fetchAnnouncements } from "./logic";
+
+type Context = {
```

Comment:
I think we really want to find another name. "Context" collides with react's Context naming, and here it's not even the "context" provided by your AnnouncementsContext.

I usually call this "user settings". like in CountervaluesUserSettings.